dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/SQLServerPreparedStatementExcecution.java

```
@@ -0,0 +1,96 @@
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(InstrumenterModule.class)
+public final class SQLServerPreparedStatementExcecution extends InstrumenterModule.Tracing
+    implements Instrumenter.ForBootstrap, Instrumenter.ForTypeHierarchy {
+
+  public SQLServerPreparedStatementExcecution() {
+    super("jdbc");
+  }
+
+  @Override
+  public String hierarchyMarkerType() {
+    return null; // bootstrap type
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> hierarchyMatcher() {
+    return implementsInterface(named("java.sql.PreparedStatement"));
+  }
+
+  @Override
+  public Map<String, String> contextStore() {
+    return singletonMap("java.sql.Connection", DBInfo.class.getName());
+  }
+
+  @Override
+  public void methodAdvice(MethodTransformer transformer) {
+    transformer.applyAdvice(
+        nameStartsWith("execute").and(takesArguments(0)).and(isPublic()),
+        SQLServerPreparedStatementExcecution.class.getName()
+            + "$SQLServerPreparedStatementExcecutionAdvice");
+  }
+
+  public static class SQLServerPreparedStatementExcecutionAdvice {
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static AgentScope onEnter(@Advice.This final PreparedStatement statement) {
+      try {
+        final Connection connection = statement.getConnection();
+        final AgentSpan span = startSpan(DATABASE_QUERY);
+        Statement instrumentationStatement = connection.createStatement();
+        instrumentationStatement.execute(
+            "set context_info 0x"
+                + span.getTraceId().toHexString()
+                + DDSpanId.toHexStringPadded(span.getSpanId())
+                + "0");
+        instrumentationStatement.close();
+
+        return activateSpan(span);
+      } catch (SQLException e) {
+        // if we can't get the connection for any reason
+        return null;
+      }
```

I don't think you need this, because of the `@Advice.OnMethodEnter(suppress = Throwable.class)` annotation which is going to catch all exceptions anyway.